src/virtual-pet.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const MINIMUM_AGE = 0;
+const MAXIMUM_AGE = 30;
+const MINIMUM_HUNGER = 0;
+const MAXIMUM_HUNGER = 10;
+const MINIMUM_FITNESS = 0;
+const MAXIMUM_FITNESS = 10;
+const FITNESS_THRESHOLD = 3;
+const HUNGER_THRESHOLD = 5;
+const ERROR_MESSAGE = 'Your pet is no longer alive :(';
+
+function Pet(name) {
+    this.name = name;
+    this.age = MINIMUM_AGE;
+    this.hunger = MINIMUM_HUNGER;
+    this.fitness = MAXIMUM_FITNESS;
+    this.children = [];
+
+    Object.defineProperty(this, 'isAlive', {
+        get: function () {
+            return this.age < MAXIMUM_AGE && this.hunger < MAXIMUM_HUNGER && this.fitness > MINIMUM_FITNESS;
+        }
+    });
+}
+
+Pet.prototype.growUp = function () {
+    if (!this.isAlive) throw new Error(ERROR_MESSAGE);
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness -= 3;
+}
+
+Pet.prototype.walk = function () {
+    if (!this.isAlive) throw new Error(ERROR_MESSAGE);
+    this.fitness + 4 > MAXIMUM_FITNESS ? this.fitness = 10 : this.fitness += 4;
```

Comment:
A (slightly) cleaner alternative to this would be:
```suggestion
    this.fitness = Math.min(MAXIMUM_FITNESS, this.fitness + 4)
```

The same structure could also be applied to the feed method